middleware/auth/auth.go

```diff
@@ -0,0 +1,45 @@
+package auth
+
+import (
+	"time"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/SermoDigital/jose/jws"
+	"github.com/SermoDigital/jose/crypto"
+)
+
+//Leeways for token expiration and token not-before times. Default to 5 minutes
+var expLeeway int64 = 300000000000
+var nbfLeeway int64 = 300000000000
+
+type authHandler struct {
+	handler http.Handler
+}
+
+func RequireAuthentication() mux.MiddlewareFunc {
```

Comment:
Also add an authentication handler that requires them to be not-logged in. We can use that to "protect" the login screen so people that are logged in can't go to the login or create account screen.

Reply:
This is optional.